### PR TITLE
Added signature verification support

### LINKED ISSUE #247: image admission: block uploading images to zot if signature verification fails
zot is cosign-compatible [1].
However zot is only a transit point for images (accept all images) and signing and verification is left to pushers and pullers.
Can we setup zot so that if can reject images which fail verification? This just needs the public key of the signers.

[1] https://github.com/sigstore/cosign

### PR DESCRIPTION
**Pull request by rafaeljigau**
Signed-off-by: Jigau Iosif-Rafael <ijigau@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:  fixes #247 

**What does this PR do / Why do we need it**: Verify images signature periodically.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/storage_fs.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ const (
 	// BlobUploadDir defines the upload directory for blob uploads.
 	BlobUploadDir = ".uploads"
 	SchemaVersion = 2
-	gcDelay       = 1 * time.Hour
+	gcDelay       = 1 * time.Nanosecond
```
Comment (rchincha) [inline pkg/storage/storage_fs.go:37]: This can't be right! 
You can add another delay timer for this.

File: Makefile [hunk 2]
```diff
@@ -19,7 +19,7 @@ binary-minimal: swagger
 
 .PHONY: binary
 binary: swagger
-	env CGO_ENABLED=0 GOOS=$(OS) GOARCH=$(ARCH) go build -o bin/zot -tags extended,containers_image_openpgp -v -trimpath -ldflags "-X  github.com/project-zot/zot/pkg/api/config.Commit=${COMMIT} -X github.com/project-zot/zot/pkg/api/config.BinaryType=extended -X github.com/project-zot/zot/pkg/api/config.GoVersion=${GO_VERSION} -s -w" ./cmd/zot
+	env CGO_ENABLED=0 GOOS=$(OS) GOARCH=$(ARCH) go build -o bin/zot -tags extended,containers_image_openpgp -v -trimpath -ldflags "-X  github.com/project-zot/zot/pkg/api/config.Commit=${COMMIT} -X github.com/project-zot/zot/pkg/api/config.BinaryType=extended -X github.com/project-zot/zot/pkg/api/config.GoVersion=${GO_VERSION}" ./cmd/zot
```
Comment (rchincha) [inline Makefile:22]: rebase, this change doesn't look right.

File: examples/config-sign.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "http": {
+        "address": "127.0.0.1",
+        "port": "8080",
+        "ReadOnly": false
+    },
+    "log": {
+        "level": "debug"
+    },
+	"extensions": {
+		"sign": {
```
Comment (rchincha) [inline examples/config-sign.json:15]: call this extension "signature"

File: examples/config-sign.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            "enable": false,
+            "verificationInterval": "1m",
+            "publicKeys":{
```
Comment (rchincha) [inline examples/config-sign.json:18]: "publicKeys" at this level are global applicable for all repos and a list.

File: examples/config-sign.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                "alpine": "/home/ijigau/Downloads/pass1/cosign.pub",
+                "repo": "keypath"
+            },
+            "publicKeys3": [
```
Comment (rchincha) [inline examples/config-sign.json:22]: Make this a per-repo config (invert the structure)
"repositories": [{
"prefix":<>,
"publicKey": <>,
},
{
...
}
]

File: pkg/storage/storage_fs.go (hunk 1 above)
Comment (rafaeljigau) [reply pkg/storage/storage_fs.go:37]: The problem is that I have to use the function ifOlderThan(), which is non exported, so I can't use it as I would wish. 
So this was just a temporary workaround to check if my code actually deletes blobs.

File: Makefile (hunk 2 above)
Comment (rafaeljigau) [reply Makefile:22]: definetly, changed so I can debug in delve.

**Comment by andaaron** [comment]
Will pick this up later.

**Pull request closed without merging**